components/public-api-server/pkg/apiv1/workspace_test.go

```diff
@@ -26,72 +27,60 @@ import (
 )
 
 func TestWorkspaceService_GetWorkspace(t *testing.T) {
-	const (
-		bearerToken      = "bearer-token-for-tests"
-		foundWorkspaceID = "easycz-seer-xl8o1zacpyw"
-	)
 
-	type Expectation struct {
-		Code     connect.Code
-		Response *v1.GetWorkspaceResponse
-	}
+	workspaceID := workspaceTestData[0].Protocol.Workspace.ID
 
-	scenarios := []struct {
-		name        string
-		WorkspaceID string
-		Workspaces  map[string]protocol.WorkspaceInfo
-		Expect      Expectation
-	}{
-		{
-			name:        "returns a workspace when workspace is found by ID",
-			WorkspaceID: foundWorkspaceID,
-			Workspaces: map[string]protocol.WorkspaceInfo{
-				foundWorkspaceID: workspaceTestData[0].Protocol,
-			},
-			Expect: Expectation{
-				Response: &v1.GetWorkspaceResponse{
-					Result: workspaceTestData[0].API,
-				},
-			},
-		},
-		{
-			name:        "not found when workspace is not found by ID",
-			WorkspaceID: "some-not-found-workspace-id",
-			Expect: Expectation{
-				Code: connect.CodeNotFound,
-			},
-		},
-	}
+	t.Run("invalid argument when workspace ID is missing", func(t *testing.T) {
```

Comment:
Changing the tests into individual sub-tests made it much easier to follow the test and debug.